ua/node_id.go

```diff
@@ -118,10 +142,26 @@ func ParseNodeID(s string) (*NodeID, error) {
 	}
 
 	// parse namespace
-	var ns uint16
+	var nsid uint16
+	var nsu string
 	switch {
 	case strings.HasPrefix(nsval, "nsu="):
-		return nil, errors.Errorf("namespace urls are not supported: %s", s)
+		if ns == nil {
+			return nil, errors.Errorf("namespace urls require a server NamespaceArray")
+		}
+
+		nsuval := strings.TrimPrefix(nsval, "nsu=")
+		ok := false
+		for id, uri := range ns {
+			if uri == nsuval {
+				nsid, nsu = uint16(id), uri
+				ok = true
+				break
+			}
+		}
+		if !ok {
+			return nil, errors.Errorf("namespace uri %s not found in the server NamespaceArray %s", nsval, ns)
```

Comment:
nit: @siscia suggested using `%q`. makes sense, though we aren't particularly consistent about this.

Reply:
No, we're not

Reply:
When I look at the `ParseNodeID` function we're always using `%s` - right or not. I prefer consistency here. If we want to change this we should change it everywhere or at least in enough places.